worlds/yoshisisland/Locations.py

```diff
@@ -0,0 +1,352 @@
+    LocationData("Raphael The Raven's Boss Room", "Raphael The Raven's Boss Room", None, 0x37, lambda state: logic._58Boss(state)),
+    ######################################################################################################
+
+    LocationData("6-1", "Scary Skeleton Goonies!: Red Coins", 0x3050C1, 0x3C),
+    LocationData("6-1", "Scary Skeleton Goonies!: Flowers", 0x3050C2, 0x3C),
+    LocationData("6-1", "Scary Skeleton Goonies!: Stars", 0x3050C3, 0x3C),
+    LocationData("6-1", "Scary Skeleton Goonies!: Level Clear", 0x3050C4, 0x3C),
+
+    LocationData("6-2", "The Cave Of The Bandits: Red Coins", 0x3050C5, 0x3D),
+    LocationData("6-2", "The Cave Of The Bandits: Flowers", 0x3050C6, 0x3D),
+    LocationData("6-2", "The Cave Of The Bandits: Stars", 0x3050C7, 0x3D),
+    LocationData("6-2", "The Cave Of The Bandits: Level Clear", 0x3050C8, 0x3D),
+
+    LocationData("6-3", "Beware The Spinning Logs: Red Coins", 0x3050C9, 0x3E),
+    LocationData("6-3", "Beware The Spinning Logs: Flowers", 0x3050CA, 0x3E),
+    LocationData("6-3", "Beware The Spinning Logs: Stars", 0x3050CB, 0x3E),
+    LocationData("6-3", "Beware The Spinning Logs: Level Clear", 0x3050CC, 0x3E),
+
+    LocationData("6-4", "Tap-Tap The Red Nose's Fort: Red Coins", 0x3050CD, 0x3F),
+    LocationData("6-4", "Tap-Tap The Red Nose's Fort: Flowers", 0x3050CE, 0x3F),
+    LocationData("6-4", "Tap-Tap The Red Nose's Fort: Stars", 0x3050CF, 0x3F),
+    LocationData("6-4", "Tap-Tap The Red Nose's Fort: Level Clear", 0x3050D0, 0x3F, lambda state: logic._64CanFightBoss(state)),
+    LocationData("Tap-Tap The Red Nose's Boss Room", "Tap-Tap The Red Nose's Boss Room", None, 0x3F, lambda state: logic._64Boss(state)),
+
+    LocationData("6-5", "The Very Loooooong Cave: Red Coins", 0x3050D1, 0x40),
+    LocationData("6-5", "The Very Loooooong Cave: Flowers", 0x3050D2, 0x40),
+    LocationData("6-5", "The Very Loooooong Cave: Stars", 0x3050D3, 0x40),
+    LocationData("6-5", "The Very Loooooong Cave: Level Clear", 0x3050D4, 0x40),
+
+    LocationData("6-6", "The Deep, Underground Maze: Red Coins", 0x3050D5, 0x41),
+    LocationData("6-6", "The Deep, Underground Maze: Flowers", 0x3050D6, 0x41),
+    LocationData("6-6", "The Deep, Underground Maze: Stars", 0x3050D7, 0x41),
+    LocationData("6-6", "The Deep, Underground Maze: Level Clear", 0x3050D8, 0x41),
+
+    LocationData("6-7", "KEEP MOVING!!!!: Red Coins", 0x3050D9, 0x42),
+    LocationData("6-7", "KEEP MOVING!!!!: Flowers", 0x3050DA, 0x42),
+    LocationData("6-7", "KEEP MOVING!!!!: Stars", 0x3050DB, 0x42),
+    LocationData("6-7", "KEEP MOVING!!!!: Level Clear", 0x3050DC, 0x42),
+
+    LocationData("6-8", "King Bowser's Castle: Red Coins", 0x3050DD, 0x43),
+    LocationData("6-8", "King Bowser's Castle: Flowers", 0x3050DE, 0x43),
+    LocationData("6-8", "King Bowser's Castle: Stars", 0x3050DF, 0x43)
+    )
+
+    if not world or world.options.extras_enabled.value == 1:
+        location_table += ( 

```

Comment:
these should be lists not tuples. += on tuples creates completely new objects and is incredibly slow.

Reply:
Would it be faster if they weren't += if the setting was on, but instead removed if the setting is off?

Reply:
no they should just be lists